packages/web/src/components/track/GiantTrackTile.js

```diff
@@ -303,22 +314,66 @@ class GiantTrackTile extends PureComponent {
     )
   }
 
+  renderProgressInfo() {
+    const { duration, playbackPositionInfo } = this.props
+    if (playbackPositionInfo) {
+      if (playbackPositionInfo.status === 'IN_PROGRESS') {
+        const remainingTime = duration - playbackPositionInfo.playbackPosition
+        return (
+          <div>
+            <p className={styles.progressText}>
+              {`${formatLineupTileDuration(remainingTime, true)} ${
+                messages.timeLeft
+              }`}
+            </p>
+            <ProgressBar
+              value={(playbackPositionInfo.playbackPosition / duration) * 100}
+              sliderClassName={styles.progressTextSlider}
+            />
+          </div>
+        )
+      } else if (playbackPositionInfo.status === 'COMPLETED') {
+        return (
+          <div>
+            <p className={styles.completeText}>
+              {messages.played}
+              <IconCheck className={styles.completeTextIcon} />
+            </p>
+          </div>
+        )
+      }
+    } else {
+      return <p className={styles.progressText}>{messages.unplayed}</p>
+    }
+  }
+
   renderStatsRow() {
     const {
       isUnlisted,
+      genre,
       repostCount,
       saveCount,
       onClickReposts,
       onClickFavorites
     } = this.props
+    const isPodcast = genre === Genre.PODCASTS
+    const isNewPodcastControlsEnabled = getFeatureEnabled(
+      FeatureFlags.PODCAST_CONTROL_UPDATES_ENABLED
+    )
+
     return (
-      <RepostFavoritesStats
-        isUnlisted={isUnlisted}
-        repostCount={repostCount}
-        saveCount={saveCount}
-        onClickReposts={onClickReposts}
-        onClickFavorites={onClickFavorites}
-      />
+      <>
+        <RepostFavoritesStats
+          isUnlisted={isUnlisted}
+          repostCount={repostCount}
+          saveCount={saveCount}
+          onClickReposts={onClickReposts}
+          onClickFavorites={onClickFavorites}
+        />
+        {isPodcast && isNewPodcastControlsEnabled
+          ? this.renderListenCount()
```

Comment:
then one nice addition is that we can just render <Component /> here instead of renderFn